src/app/api/delegates/getDelegates.ts

```diff
@@ -62,19 +62,23 @@ export async function getDelegates({
     pageSize
   );
 
-  const statements = await Promise.all(
-    delegates.map((delegate) =>
-      getDelegateStatement({ addressOrENSName: delegate.delegate })
-    )
+  const _delegates = await Promise.all(
+    delegates.map(async (delegate) => {
+      return {
+        citizen: await isCitizen(delegate.delegate),
```

Comment:
out of curiosity, wondering if you tried to throw this directly in the query?

it's still very performant, so don't think there's a need